Source filepath cannot contain spaces

## Describe the bug
When providing a directory path with spaces as the source directory, the script fails when running the `opj_compress` command. This happens when providing the directory path  in double quotes or escaping the spaces using a backslash.

## To reproduce
Run the script over a directory that contains spaces in its path.

## Expected behavior
Paths for source directories can include spaces.

## Error message or screenshots
```command '['/usr/local/bin/opj_compress -i /mnt/xdrive/Projects/Reference Digitization/Uploaded_IIIF/94a9865034f8430bb4efe857825c4012/master/94a9865034f8430bb4efe857825c4012_002.tif -o tmp/images/VUDsYHErtMTUPkb7Lb8VWQ_002.jp2 -n 6 -r 1.5 -c [256,256],[256,256],[128,128] -b 64,64 -p RPCL -SOP']' returned with error (code 1): b'[ERROR] Unknown input file format: /mnt/xdrive/Projects/Reference \n        Known file formats are *.pnm, *.pgm, *.ppm, *.pgx, *png, *.bmp, *.tif, *.raw or *.tga\n'```

## Impact on your work
Current workaround is to move the source directory to a path that does not contain spaces, so this is not a blocker.

## Additional context
It appears that this issue is happening when using `subprocess` in the `create_jp2` method. The `os.path.isdir` check in `pipeline.py` is passing before the subprocess command that's throwing the error.

--- a/README.md
+++ b/README.md
@@ -14,7 +14,7 @@ Then build the image:
     $ docker build . -t iiif-pipeline
 
 You can then run the pipeline, mounting any local directories you need access to.
-For example, the command above will mount `/local_files/` on your local machine
+For example, the command below will mount `/local_files/` on your local machine
 to `/source_files/` in the running container and then execute `iiif_pipeline.py`,
 using `/tmp` in the container as a destination directory:
 
@@ -68,6 +68,10 @@ created before they are uploaded. The user running the script must own the
 `target_directory`. The optional `--skip` flag will skip image files with
 filenames ending in `_001`, and the optional `--replace` flag will replace
 existing files.
+
+If the path to the `source_directory` or `target_directory` include spaces, you must wrap them in either single or double quotation marks:
+
+  $ iiif-pipeline.py 'source directory' 'target directory' [--skip] [--replace]
 
 
 ## Configuration

--- a/iiif_pipeline/derivatives.py
+++ b/iiif_pipeline/derivatives.py
@@ -70,7 +70,6 @@ def create_jp2(files, identifier, derivative_dir, replace=False):
                        "-o", derivative_path,
                        "-n", str(layers),
                        "-SOP"] + default_options
-                print(cmd)
                 subprocess.run(cmd)
             else:
                 raise Exception(

--- a/iiif_pipeline/derivatives.py
+++ b/iiif_pipeline/derivatives.py
@@ -65,11 +65,13 @@ def create_jp2(files, identifier, derivative_dir, replace=False):
         else:
             if is_tiff(original_file):
                 layers = calculate_layers(original_file)
-                subprocess.run(["/usr/local/bin/opj_compress",
-                                "-i", original_file,
-                                "-o", derivative_path,
-                                "-n", str(layers),
-                                "-SOP"] + default_options)
+                cmd = ["/usr/local/bin/opj_compress",
+                       "-i", original_file,
+                       "-o", derivative_path,
+                       "-n", str(layers),
+                       "-SOP"] + default_options
+                print(cmd)
+                subprocess.run(cmd)
             else:
                 raise Exception(
                     "Error creating JPEG2000: {} is not a valid TIFF".format(original_file))

--- a/setup.py
+++ b/setup.py
@@ -15,7 +15,7 @@ setup(
     license='MIT',
     packages=find_packages(),
     install_requires=[
-        'ArchivesSnake'
+        'ArchivesSnake',
         'boto3',
         'iiif-prezi',
         'img2pdf',

--- a/tests/test_create_jp2.py
+++ b/tests/test_create_jp2.py
@@ -8,34 +8,38 @@ from iiif_pipeline.derivatives import create_jp2
 from iiif_pipeline.helpers import matching_files
 
 FIXTURES_FILEPATH = os.path.join("fixtures", "tif")
-SOURCE_DIR = os.path.join("/", "source")
-DERIVATIVE_DIR = os.path.join("/", "derivatives")
+SOURCES = [os.path.join("/", "source"),
+           os.path.join("/", "source folder")]
+DERIVATIVES = [os.path.join("/", "derivatives"),
+               os.path.join("/", "derivatives folder")]
 UUIDS = [random_string() for x in range(random.randint(2, 3))]
 PAGE_COUNT = random.randint(1, 5)
 
 
 def setup():
     """Sets up source directory for tests."""
-    for d in [SOURCE_DIR, DERIVATIVE_DIR]:
-        if os.path.isdir(d):
-            shutil.rmtree(d)
-    shutil.copytree(FIXTURES_FILEPATH, SOURCE_DIR)
-    copy_sample_files(SOURCE_DIR, UUIDS, PAGE_COUNT, "tif")
-    os.makedirs(DERIVATIVE_DIR)
+    for SOURCE_DIR, DERIVATIVE_DIR in zip(SOURCES, DERIVATIVES):
+        for d in [SOURCE_DIR, DERIVATIVE_DIR]:
+            if os.path.isdir(d):
+                shutil.rmtree(d)
+        shutil.copytree(FIXTURES_FILEPATH, SOURCE_DIR)
+        copy_sample_files(SOURCE_DIR, UUIDS, PAGE_COUNT, "tif")
+        os.makedirs(DERIVATIVE_DIR)
 
 
 def test_create_jp2():
     """Ensure the run method produces the expected number of files."""
-    uuid = random.choice(UUIDS)
-    tiff_files = matching_files(
-        SOURCE_DIR,
-        prefix=uuid,
-        skip=False,
-        prepend=True)
-    create_jp2(tiff_files, uuid, DERIVATIVE_DIR)
-    assert len(os.listdir(DERIVATIVE_DIR)) == PAGE_COUNT
-    for f in os.listdir(DERIVATIVE_DIR):
-        assert ".tif" not in f
+    for SOURCE_DIR, DERIVATIVE_DIR in zip(SOURCES, DERIVATIVES):
+        uuid = random.choice(UUIDS)
+        tiff_files = matching_files(
+            SOURCE_DIR,
+            prefix=uuid,
+            skip=False,
+            prepend=True)
+        create_jp2(tiff_files, uuid, DERIVATIVE_DIR)
+        assert len(os.listdir(DERIVATIVE_DIR)) == PAGE_COUNT
+        for f in os.listdir(DERIVATIVE_DIR):
+            assert ".tif" not in f
 
 
 def test_replace_jp2():
@@ -44,19 +48,21 @@ def test_replace_jp2():
     If files exist and the replace parameter is False, a FileExistsError should
     be raised.
     """
-    uuid = random.choice(UUIDS)
-    tiff_files = matching_files(
-        SOURCE_DIR,
-        prefix=uuid,
-        skip=False,
-        prepend=True)
-    create_jp2(tiff_files, uuid, DERIVATIVE_DIR)
-    with pytest.raises(FileExistsError):
+    for SOURCE_DIR, DERIVATIVE_DIR in zip(SOURCES, DERIVATIVES):
+        uuid = random.choice(UUIDS)
+        tiff_files = matching_files(
+            SOURCE_DIR,
+            prefix=uuid,
+            skip=False,
+            prepend=True)
         create_jp2(tiff_files, uuid, DERIVATIVE_DIR)
-    create_jp2(tiff_files, uuid, DERIVATIVE_DIR, replace=True)
+        with pytest.raises(FileExistsError):
+            create_jp2(tiff_files, uuid, DERIVATIVE_DIR)
+        create_jp2(tiff_files, uuid, DERIVATIVE_DIR, replace=True)
 
 
 def teardown():
     """Removes test directories."""
-    for d in [SOURCE_DIR, DERIVATIVE_DIR]:
-        shutil.rmtree(d)
+    for SOURCE_DIR, DERIVATIVE_DIR in zip(SOURCES, DERIVATIVES):
+        for d in [SOURCE_DIR, DERIVATIVE_DIR]:
+            shutil.rmtree(d)

